docs/rpc-endpoints.md

```diff
@@ -0,0 +1,255 @@
+          "type": "uint128"
+        }
+      ],
+      "outputs": {
+        "type": "bool"
+      }
+    },
+    {
+      "name": "update-info",
+      "access": "public",
+      "args": [],
+      "outputs": {
+        "type": {
+          "response": {
+            "ok": "bool",
+            "error": "none"
+          }
+        }
+      }
+    },
+    {
+      "name": "get-exotic-data-info",
+      "access": "read_only",
+      "args": [
+        {
+          "name": "height",
+          "type": "uint128"
+        }
+      ],
+      "outputs": {
+        "type": {
+          "tuple": [
+            {
+              "name": "btc-hash",
+              "type": {
+                "buffer": {
+                  "length": 32
+                }
+              }
+            },
+            {
+              "name": "burn-block-time",
+              "type": "uint128"
+            },
+            {
+              "name": "id-hash",
+              "type": {
+                "buffer": {
+                  "length": 32
+                }
+              }
+            },
+            {
+              "name": "stacks-hash",
+              "type": {
+                "buffer": {
+                  "length": 32
+                }
+              }
+            },
+            {
+              "name": "stacks-miner",
+              "type": "principal"
+            },
+            {
+              "name": "vrf-seed",
+              "type": {
+                "buffer": {
+                  "length": 32
+                }
+              }
+            }
+          ]
+        }
+      }
+    }
+  ],
+  "variables": [],
+  "maps": [
+    {
+      "name": "block-data",
+      "key": [
+        {
+          "name": "height",
+          "type": "uint128"
+        }
+      ],
+      "value": [
+        {
+          "name": "btc-hash",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        },
+        {
+          "name": "burn-block-time",
+          "type": "uint128"
+        },
+        {
+          "name": "id-hash",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        },
+        {
+          "name": "stacks-hash",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        },
+        {
+          "name": "stacks-miner",
+          "type": "principal"
+        },
+        {
+          "name": "vrf-seed",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        }
+      ]
+    }
+  ],
+  "fungible_tokens": [],
+  "non_fungible_tokens": []
+}
+```
+
+### GET /v2/contracts/source/[Stacks Address]/[Contract Name]
+
+Fetch the source for a smart contract. Returned as a JSON string.
+
+### POST /v2/contracts/call-read/[Stacks Address]/[Contract Name]/[Function Name]
+
+Call a read-only public function on a given smart contract.
```

Comment:
Should you provide a block-height when hitting this endpoint?

Reply:
I think it would be better to provide a block hash, and provide separate API endpoints for:

* Getting the block hash of the canonical chain tip
* Getting the block hash at a particular height, given a chain tip.

The reason I'm against using a block height is because there are potentially many different blocks at the same height, but each block has its own index hash (i.e. hash(burn-header-hash + stacks-block-hash)).

Reply:
Agreed on adding block hash specified endpoints -- I think we should have block hash endpoints, but probably not block height specified endpoints -- the advantage to the block hash endpoints is that their returns are actually constant (once they've processed the block in question).

Reply:
Pardon my lack of precision, by block-height I meant the notion of height / anchoring the request to a specific block. Sounds good to me!